packages/invoices/src/OutboundShipment/DetailView/modals/ItemDetailsModal.tsx

```diff
@@ -0,0 +1,239 @@
+import React, { SyntheticEvent } from 'react';
+
+import {
+  DialogButton,
+  Grid,
+  InvoiceLine,
+  Item,
+  gql,
+  request,
+  useForm,
+  useQuery,
+  useDialog,
+  FormProvider,
+} from '@openmsupply-client/common';
+import { Environment } from '@openmsupply-client/config';
+import { BatchesTable } from './BatchesTable';
+import { ItemDetailsForm } from './ItemDetailsForm';
+import { BatchRow } from '../types';
+
+interface ItemDetailsModalProps {
+  invoiceLine?: InvoiceLine;
+  isOpen: boolean;
+  onClose: () => void;
+  upsertInvoiceLine: (invoiceLine: InvoiceLine) => void;
+}
+
+export const getInvoiceLine = (
+  id: string,
+  item: Item,
+  line: { id: string; expiryDate: string },
+  quantity: number
+): InvoiceLine => ({
+  id,
+  itemName: item.name,
+  stockLineId: line.id,
+  itemCode: item.code,
+  quantity,
+  invoiceId: '',
+  expiry: line.expiryDate,
+});
+
+const listQueryFn = async (): Promise<Item[]> => {
+  const { items } = await request(
+    Environment.API_URL,
+    gql`
+      query items {
+        items {
+          data {
+            id
+            isVisible
+            name
+            code
+            availableQuantity
+            availableBatches {
+              nodes {
+                id
+                batch
+                expiryDate
+                packSize
+                costPricePerPack
+                sellPricePerPack
+                availableNumberOfPacks
+                totalNumberOfPacks
+              }
+            }
+          }
+        }
+      }
+    `
+  );
+
+  return items.data;
+};
+
+const sortByExpiryDate = (a: BatchRow, b: BatchRow) => {
+  const expiryA = new Date(a.expiryDate);
+  const expiryB = new Date(b.expiryDate);
+
+  if (expiryA < expiryB) {
+    return -1;
+  }
+  if (expiryA > expiryB) {
+    return 1;
+  }
+
+  return 0;
+};
+
+export const ItemDetailsModal: React.FC<ItemDetailsModalProps> = ({
+  invoiceLine,
+  isOpen,
+  onClose,
+  upsertInvoiceLine,
+}) => {
+  const [batchRows, setBatchRows] = React.useState<BatchRow[]>([]);
+  const [selectedItem, setSelectedItem] = React.useState<Item | null>(null);
+  const [quantity, setQuantity] = React.useState(0);
+  const [isAllocated, setIsAllocated] = React.useState(false);
+  const [lines, setLines] = React.useState<InvoiceLine[]>(
+    invoiceLine ? [invoiceLine] : []
+  );
+  const { hideDialog, showDialog, Modal } = useDialog({
+    title: 'heading.add-item',
+  });
+  const methods = useForm({ mode: 'onBlur' });
```

Comment:
`onBlur` to trigger the form validation. However in the end I've removed the form validation - it wasn't reporting `isDirty` when it was dirty and the `isValid` was always true - and then it wouldn't allow the form to submit 🙄 
Obvs I'm doing something wrong, but didn't see what it was.
Ignoring the form and using explicit event methods and local state has been more reliable